airbyte-integrations/base-java/src/main/java/io/airbyte/integrations/base/IntegrationRunner.java

```diff
@@ -0,0 +1,95 @@
+/*
+ * MIT License
+ *
+ * Copyright (c) 2020 Airbyte
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+
+package io.airbyte.integrations.base;
+
+import com.fasterxml.jackson.databind.JsonNode;
+import io.airbyte.commons.io.IOs;
+import io.airbyte.commons.json.Jsons;
+import io.airbyte.config.Schema;
+import io.airbyte.singer.SingerMessage;
+import java.nio.file.Path;
+import java.util.Optional;
+import java.util.Scanner;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class IntegrationRunner {
+
+  private static final Logger LOGGER = LoggerFactory.getLogger(IntegrationRunner.class);
+
+  private final Destination destination;
+
+  public IntegrationRunner(Destination destination) {
+    this.destination = destination;
+  }
+
+  public void run(String[] args) throws Exception {
+    final IntegrationConfig parsed = new IntegrationCliParser().parse(args);
+
+    switch (parsed.getCommand()) {
+      case SPEC -> System.out.println(Jsons.serialize(destination.spec()));
+      case CHECK -> {
+        final JsonNode config = parseConfig(parsed.getConfig());
+        System.out.println(Jsons.serialize(destination.check(config)));
```

Comment:
I don't think PMD will let that compile

Reply:
actually it's not complaining. not sure why tbh.